processor/labelsprocessor/labels_processor.go

```diff
@@ -0,0 +1,124 @@
+package labelsprocessor
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	v11 "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/common/v1"
+	v1 "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/metrics/v1"
+)
+
+type labelMetricProcessor struct {
+	cfg *Config
+}
+
+func newLabelMetricProcessor(cfg *Config) (*labelMetricProcessor, error) {
+	err := validateConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &labelMetricProcessor{cfg: cfg}, nil
+}
+
+func validateConfig(cfg *Config) error {
+	// Ensure no empty keys/values exist
+	for _, elem := range cfg.Labels {
+		if elem.Key == "" || elem.Value == "" {
```

Comment:
What if value is `null`? Does the underlying parser converts `null` to empty string?

Reply:
I will double check this and change if necessary

Reply:
I can confirm from testing that all invalid configurations are handled with this logic, I will also add a new test case to verify that `nil` field values are handled